ExampleMod/Content/NPCs/ExampleWorm.cs

```diff
@@ -49,6 +52,13 @@ public override void SetBestiary(BestiaryDatabase database, BestiaryEntry bestia
 			});
 		}
 
+		public override void OnKill() {
+			ModContent.GetModAchievement<ExampleWormKilled>().GetCondition("EXAMPLEMOD_KILL_BOOLEAN_CONDITION").Complete();
+			((CustomIntCondition)ModContent.GetModAchievement<ManyExampleWormsKilled>().GetCondition("EXAMPLEMOD_KILL_WORMS")).Value++;
```

Comment:
As with the advice above, the cast would not be needed if the `CustomIntCondition Condition` field could be accessed directly.